builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -455,6 +455,168 @@ func TestAccVSphereVirtualMachine_createWithCdrom(t *testing.T) {
 	})
 }
 
+func TestAccVSphereVirtualMachine_updateMemory(t *testing.T) {
+	var vm virtualMachine
+	var locationOpt string
+	var datastoreOpt string
+
+	if v := os.Getenv("VSPHERE_DATACENTER"); v != "" {
```

Comment:
With a vSphere server you can maintain multiple vSphere data centers. Below the data centers you usually have clusters containing physical hosts and resource pools. So you have to specify the data center at creation time of the VM since it's part of the location of the VM. Using (offline?) Vmotion you can move the VM from one data center to another.